libc/src/__support/CPP/type_traits/invoke_result.h

```diff
@@ -0,0 +1,26 @@
+//===-- invoke_result type_traits -------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifndef LLVM_LIBC_SRC_SUPPORT_CPP_TYPE_TRAITS_INVOKE_RESULT_H
+#define LLVM_LIBC_SRC_SUPPORT_CPP_TYPE_TRAITS_INVOKE_RESULT_H
+
+#include "src/__support/CPP/type_traits/invoke.h"
+#include "src/__support/CPP/utility/declval.h"
+
+namespace __llvm_libc::cpp {
+
+template <class F, class... Args> struct invoke_result {
+  using type =
+      decltype(cpp::invoke(cpp::declval<F>(), cpp::declval<Args>()...));
```

Comment:
`cpp::invoke(cpp::declval<F>()` always creates a `F&&`, so what happens if we have a `&&`-qualified `operator()` and and a `&`-qualified `operator()` with different return types ?

Can you add a test for 

```
struct Functor {
  int operator()() &&;
  double operator()() &;
};
```